### ISSUE TITLE
`scripted` parallel execution causing issues on slow machines

### ISSUE THREAD

**Issue by ptrdom** [sbt/sbt#7431]
## steps

- Have a sbt project with multiple modules of plugins defined.
- Have `scripted` tests with multiple dependencies defined in their build definitions for each of the plugin modules.
- Run `scripted` in a slow machine - Windows CI runner with no cache present is a good example case.
- `scripted` frequently fails with errors like mentioning this - `C:\Users\runneradmin\AppData\Local\Coursier\Cache\v1\.structure.lock (The process cannot access the file because it is being used by another process)` 

## problem

I think the issue is in the fact that `scripted` executes in parallel between sbt modules, that leads to multiple sbt instances being spawned in parallel that share single Coursier cache, which leads to illegal concurrent access exceptions.

## expectation

`scripted` should either work fine with parallel per module execution or have simple means of disabling current parallel behavior. I tried adding a tag for concurrent restrictions to already existing `scripted` task to limit its concurrent to `1` and failed to do so, maybe it is possible and it is just an error on my end.

## notes

Example case - https://github.com/ptrdom/scalajs-esbuild/actions/runs/6849978751. All runs throw errors mentioning failure to download some different dependency.

What I ended up doing is writing a custom task that executes `scripted` sequentially per module in CI - https://github.com/ptrdom/scalajs-esbuild/commit/867cb5346fb0e0a0514de66d82bb82bc20d2c281. I am guessing that making sbt instances spawned by `scripted` use their own Coursier caches would be a working workaround too, though obviously not ideal.

**Comment by eed3si9n** [comment]
Is this a problem of scripted or Coursier? - https://github.com/coursier/coursier/issues/673

**Comment by ptrdom** [comment]
I think it is a problem of Coursier, but my guess is that it is resource intensive enough on the Coursier side to fix so that it would be simpler for sbt to have a workaround. Would having a custom `scripted` tag for `Global / concurrentRestrictions` be an option?

**Comment by eed3si9n** [comment]
Closing this as won't fix, at least from sbt side.
